tests/testsuite/build.rs

```diff
@@ -3683,6 +3683,98 @@ fn custom_target_dir() {
     );
 }
 
+#[test]
+fn custom_target_dir_line_parameter() {
+    let p = project("foo")
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+        "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    let exe_name = format!("foo{}", env::consts::EXE_SUFFIX);
+
+    assert_that(
+        p.cargo("build").arg("--target-dir").arg("foo/target"),
```

Comment:
We've recently added a nicer API for this: `p.cargo("build --target-dir foo/target")`.